lib/nexpose/alert.rb

```diff
@@ -117,135 +35,169 @@ def initialize(severity = 1, confirmed = 1, unconfirmed = 1, potential = 1)
+    # Whether or not this alert is currently active.
+    attr_accessor :enabled
+    # Send at most this many alerts per scan.
+    attr_accessor :max_alerts
+    # Alert type and its configuration. One of SMTPAlert, SyslogAlert, SNMPAlert
+    attr_accessor :alert_type
+    # Send alerts based upon scan status.
+    typed_accessor :scan_filter, ScanFilter
+    # Send alerts based upon vulnerability finding status.
+    typed_accessor :vuln_filter, VulnFilter
 
-    def initialize(server)
-      @server = server
-    end
+    # load a particular site alert
+    def self.load(nsc, site_id, alert_id)
+      uri = "/api/2.1/site_configurations/#{site_id}/alerts/#{alert_id}"
+      resp = AJAX.get(nsc, uri, AJAX::CONTENT_TYPE::JSON)
 
-    def self.parse(xml)
-      new(xml.attributes['server'])
+      unless resp.to_s == ''
+        data = JSON.parse(resp, symbolize_names: true)
+        json_initializer(data).deserialize(data)
+      end
     end
 
-    def as_xml
-      xml = REXML::Element.new('syslogAlert')
-      xml.attributes['server'] = @server
-      xml
+    # load alerts from an array of hashes
+    def self.load_alerts(alerts)
+      alerts.map { |hash| json_initializer(hash).deserialize(hash) }
     end
 
-    def to_xml
-      as_xml.to_s
+    # load a list of alerts for a given site
+    def self.list_alerts(nsc, site_id)
+      uri = "/api/2.1/site_configurations/#{site_id}/alerts"
+      resp = AJAX.get(nsc, uri, AJAX::CONTENT_TYPE::JSON)
+      data = JSON.parse(resp, symbolize_names: true)
+      self.load_alerts(data) unless data.nil?
     end
-  end
 
-  # SNMP Alert
-  # This class should only exist as an element of an Alert.
-  #
-  class SNMPAlert
+    def self.json_initializer(hash)
+      create(hash)
+    end
 
-    # The community string
-    attr_accessor :community
+    def to_h
+      to_hash(Hash.new)
+    end
 
-    # The server to sent this alert
-    attr_accessor :server
+    def to_json
+      serialize()
+    end
 
-    def initialize(community, server)
-      @community = community
-      @server = server
+    # delete an alert from the given site
+    def delete(nsc, site_id)
+      uri = "/api/2.1/site_configurations/#{site_id}/alerts/#{self.id}"
+      AJAX.delete(nsc, uri, AJAX::CONTENT_TYPE::JSON)
     end
 
-    def self.parse(xml)
-      new(xml.attributes['community'], xml.attributes['server'])
+    # save an alert for a given site
+    def save(nsc, site_id)
+      validate
+      uri = "/api/2.1/site_configurations/#{site_id}/alerts"
+      id = AJAX.put(nsc, uri, self.to_json, AJAX::CONTENT_TYPE::JSON)
+      @id = id.to_i
     end
 
-    def as_xml
-      xml = REXML::Element.new('snmpAlert')
-      xml.attributes['community'] = @community
-      xml.attributes['server'] = @server
-      xml
+    def validate()
+      raise ArgumentError.new('Name is a required attribute.') unless @name
+      raise ArgumentError.new('Scan filter is a required attribute.') unless @scan_filter
```

Comment:
Line is too long. [89/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.